R/html_dependency.R

```diff
@@ -505,6 +525,96 @@ renderDependencies <- function(dependencies,
   HTML(paste(html, collapse = "\n"))
 }
 
+
+
+renderScript <- function(script, srcpath, encodeFunc, hrefFilter) {
+  # If the input is a named list, transform it to an unnamed list
+  # whose only element is the input list
+  if (anyNamed(script)) {
+    if (anyUnnamed(script)) stop("script inputs cannot mix named and unnamed")
+    script <- list(script)
+  }
+
+  # For each element, if it's a scalar string, transform it to a named
+  # list with one element, "src".
+  script <- lapply(script, function(item) {
+    if (length(item) == 1 && is.character(item)) {
+      item = list(src = item)
+    }
+
+    if (length(names(item)) == 0) {
+      stop(
+        "Elements of script must be named lists, or scalar strings ",
+        "I got ", deparse(item)
+      )
+    }
+
+    return(item)
+  })
+
+  script <- vapply(
+    script, function(x) {
+      x$src <- hrefFilter(file.path(srcpath, encodeFunc(x$src)))
+      renderTag(x, tag_name = "script", required_attributes = "src")
```

Comment:
I'm not sure i agree, i think this could be useful if the other tag renders were modified to call this for more DRY, but I'm happy to remove it

Reply:
Yeah, I think maintaining a separate `renderTag` function isn't worth it here, especially since it's fairly straightforward to understand what the `paste` calls are doing here.